climada/util/calibrate/base.py

```diff
@@ -158,6 +159,102 @@ def plot_impf_set(self, **plot_kwargs):
         """
         return self.impf_set.plot(**plot_kwargs)
 
+    def plot_impf_variability(
+        self,
+        cost_func_diff: float = 0.1,
+        p_space_df: Optional[pd.DataFrame] = None,
```

Comment:
Can we change this around a bit? I am thinking of two cases:
1. I want the default parameter space and sample it using `cost_func_diff`. Then these default arguments make sense
2. I insert a custom parameter space. In this case, I probably want to plot the entire function space, because I already did a selection myself. But then I would have to set `cost_func_diff = np.inf`, which is very unintuitive.

What would then be a sensible default value for `cost_func_diff`? Or could we maybe add the option to set it to `None`, meaning that the entire parameter space is plotted?

Reply:
Hm, how about we keep 10% as default value and if someone adds cost_func_diff=None, we  add something like:
```
if cost_func_diff is None:
    # If no value is given, plot the complete parameter space
    max_cost_func_val = np.inf
else:
    max_cost_func_val = p_space_df['Cost Function'].min()*(1+cost_func_diff)
```
What do you think? Having 'None' as default may be confusing if someone has the complete (e.g. Bayesian) parameter space.

Furthermore, I realize that if the parameter space is given, the output doesn't have to be a BayesianOptimizerOutput anymore. So, I'll add the assertion within the "if p_space_df" code block.

Reply:
> Having 'None' as default may be confusing if someone has the complete (e.g. Bayesian) parameter space.

True, sounds good. Then you could do:

```python
if p_space_df is 
    p_space_df = self.output.p_space_to_dataframe()  # Check if that exists first!

if cost_func_diff is None:
    params_within_range = p_space_df[params]
else:
    params_within_range = p_space_df.loc[
        p_space_df['Cost Function'] <= max_cost_func_val, params
    ]
```